### ISSUE TITLE
[Bug] Galera is failing to recover after all instances shutdown

### ISSUE THREAD

**Issue by PeterStuck** [mariadb-operator/mariadb-operator#709]
**Documentation**
- [x] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**
I've been testing an operator with basic Galera setup for past couple of weeks. Now I've entered pessimistic case scenarios where my network can change or k8s cluster goes down and started to getting some problems. Currently I'm testing it locally with Microk8s.

So let's start with basics. I tried following command to simulate a Galera cluster recovery:
```bash
kubectl delete pods -l app.kubernetes.io/instance=mariadb-galera
```

This works fine for me, I didn't face any issues with recovery process. But in cases where I delete MariaDB config, shutdown k8s cluster or apply some changes on network it forces my mariadb pods to restart and seems that it's randomly whether cluster recovers fully, partially or enters infinite loop of pod restarts with the status of CrashLoopBackOff.

Let's stick with this basic Galera setup:
```bash
apiVersion: k8s.mariadb.com/v1alpha1
kind: MariaDB
metadata:
  name: mariadb-galera
spec:
  rootPasswordSecretKeyRef:
    name: mariadb
    key: root-password

  storage:
    size: 1Gi

  replicas: 3

  galera:
    enabled: true
    primary:
      podIndex: 0
      automaticFailover: true
    sst: mariabackup
    recovery:
      enabled: true
      minClusterSize: 30%
      clusterMonitorInterval: 10s
      clusterHealthyTimeout: 1m
      clusterBootstrapTimeout: 10m
      podRecoveryTimeout: 5m
      podSyncTimeout: 5m

  service:
    type: LoadBalancer
    metadata:
      annotations:
        metallb.universe.tf/loadBalancerIPs: 172.18.0.150

  primaryService:
    type: LoadBalancer
    metadata:
      annotations:
        metallb.universe.tf/loadBalancerIPs: 172.18.0.160

  secondaryService:
    type: LoadBalancer
    metadata:
      annotations:
        metallb.universe.tf/loadBalancerIPs: 172.18.0.161

  resources:
    requests:
      cpu: 256m
      memory: 256Mi
    limits:
      memory: 1Gi

  metrics:
    enabled: true
```

**Expected behaviour**

After k8s cluster/worker node shutdown, network change or reapplication of configuration the mariadb instances shall reconnect into Galera cluster, recover it's stored data and become available for other apps.

**Steps to reproduce the bug**

1. Apply manifests from examples/manifests/config
2. Apply above MariaDB kind config
3. Populate database with some schema and test data
4. Either delete and apply MariaDB config (keep PVCs created earlier), shutdown k8s cluster or apply some network changes (like a VPN client)
5. Some of instances or all of them will enter crash loop

**Debug information**

Pods:
```bash
 kubectl get pods -o wide
NAME                                                READY   STATUS             RESTARTS       AGE     IP             NODE     NOMINATED NODE   READINESS GATES
mariadb-galera-0                                    1/2     CrashLoopBackOff   5 (110s ago)   7m19s   10.1.129.115   pk-dev   <none>           <none>
mariadb-galera-1                                    1/2     CrashLoopBackOff   5 (2m5s ago)   7m19s   10.1.129.127   pk-dev   <none>           <none>
mariadb-galera-2                                    1/2     CrashLoopBackOff   5 (104s ago)   6m44s   10.1.129.70    pk-dev   <none>           <none>
mariadb-galera-init-qtwr4                           0/1     Completed          0              7m24s   10.1.129.83    pk-dev   <none>           <none>
mariadb-operator-cert-controller-5ff77bd795-g487d   1/1     Running            0              4h51m   10.1.129.84    pk-dev   <none>           <none>
mariadb-operator-dd9567c95-4c8lw                    1/1     Running            0              4h51m   10.1.129.124   pk-dev   <none>           <none>
mariadb-operator-webhook-7b79c9df99-dclcl           1/1     Running            0              4h51m   10.1.129.126   pk-dev   <none>           <none>
prometheus-operator-69cbc6f76c-ctd9q                1/1     Running            40 (8h ago)    6d7h    10.1.129.76    pk-dev   <none>           <none>
```

Events:
```bash
LAST SEEN   TYPE     REASON                 OBJECT                               MESSAGE
4m46s       Normal   NoPods                 poddisruptionbudget/mariadb-galera   No matching pods found
4m46s       Normal   SuccessfulCreate       statefulset/mariadb-galera           create Claim galera-mariadb-galera-0 Pod mariadb-galera-0 in StatefulSet mariadb-galera success
4m46s       Normal   SuccessfulCreate       statefulset/mariadb-galera           create Pod mariadb-galera-0 in StatefulSet mariadb-galera successful
4m46s       Normal   SuccessfulCreate       statefulset/mariadb-galera           create Claim storage-mariadb-galera-2 Pod mariadb-galera-2 in StatefulSet mariadb-galera success
4m46s       Normal   SuccessfulCreate       statefulset/mariadb-galera           create Claim storage-mariadb-galera-1 Pod mariadb-galera-1 in StatefulSet mariadb-galera success
4m46s       Normal   SuccessfulCreate       statefulset/mariadb-galera           create Claim galera-mariadb-galera-2 Pod mariadb-galera-2 in StatefulSet mariadb-galera success
4m46s       Normal   SuccessfulCreate       statefulset/mariadb-galera           create Claim galera-mariadb-galera-1 Pod mariadb-galera-1 in StatefulSet mariadb-galera success
4m46s       Normal   SuccessfulCreate       statefulset/mariadb-galera           create Pod mariadb-galera-2 in StatefulSet mariadb-galera successful
4m46s       Normal   SuccessfulCreate       statefulset/mariadb-galera           create Pod mariadb-galera-1 in StatefulSet mariadb-galera successful
3m29s       Normal   GaleraClusterHealthy   mariadb/mariadb-galera               Galera cluster is healthy
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-0 in StatefulSet mariadb-galera successful
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-1 in StatefulSet mariadb-galera successful
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
0s          Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
5m15s       Normal    NoPods                            poddisruptionbudget/mariadb-galera   No matching pods found
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Claim galera-mariadb-galera-0 Pod mariadb-galera-0 in StatefulSet mariadb-galera success
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-0 in StatefulSet mariadb-galera successful
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Claim storage-mariadb-galera-2 Pod mariadb-galera-2 in StatefulSet mariadb-galera success
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Claim storage-mariadb-galera-1 Pod mariadb-galera-1 in StatefulSet mariadb-galera success
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Claim galera-mariadb-galera-2 Pod mariadb-galera-2 in StatefulSet mariadb-galera success
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Claim galera-mariadb-galera-1 Pod mariadb-galera-1 in StatefulSet mariadb-galera success
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-2 in StatefulSet mariadb-galera successful
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-1 in StatefulSet mariadb-galera successful
0s          Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-2 in StatefulSet mariadb-galera successful
5m15s       Normal    GaleraClusterHealthy              mariadb/mariadb-galera               Galera cluster is healthy
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-0 in StatefulSet mariadb-galera successful
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-1 in StatefulSet mariadb-galera successful
5m15s       Warning   CalculateExpectedPodCountFailed   poddisruptionbudget/mariadb-galera   Failed to calculate the number of expected pods: found no controllers for pod "mariadb-galera-2"
5m15s       Normal    SuccessfulCreate                  statefulset/mariadb-galera           create Pod mariadb-galera-2 in StatefulSet mariadb-galera successful
```

Endpoints:
```bash
apiVersion: v1
kind: Endpoints
metadata:
  annotations:
    endpoints.kubernetes.io/last-change-trigger-time: "2024-07-01T14:15:47Z"
  creationTimestamp: "2024-07-01T14:15:12Z"
  labels:
    service.kubernetes.io/headless: ""
  name: mariadb-galera-internal
  namespace: default
  resourceVersion: "2590444"
  uid: 746f5cab-c8d7-48af-991e-4306b921f49b
subsets:
- addresses:
  - hostname: mariadb-galera-0
    ip: 10.1.129.115
    nodeName: pk-dev
    targetRef:
      kind: Pod
      name: mariadb-galera-0
      namespace: default
      uid: 05d78567-eaf3-42ab-bb21-c2f67a36e5fb
  - hostname: mariadb-galera-1
    ip: 10.1.129.127
    nodeName: pk-dev
    targetRef:
      kind: Pod
      name: mariadb-galera-1
      namespace: default
      uid: 00f7ec4f-20a7-468b-9311-a086b8be1c87
  - hostname: mariadb-galera-2
    ip: 10.1.129.70
    nodeName: pk-dev
    targetRef:
      kind: Pod
      name: mariadb-galera-2
      namespace: default
      uid: 48a6d18b-2626-4367-8cee-c2f5fbef39b5
  ports:
  - name: sst
    port: 4444
    protocol: TCP
  - name: cluster
    port: 4567
    protocol: TCP
  - name: ist
    port: 4568
    protocol: TCP
  - name: mariadb
    port: 3306
    protocol: TCP
  - name: agent
    port: 5555
    protocol: TCP
```

**Environment details**:
- Kubernetes version: [v1.29.4]
- mariadb-operator version: [v0.0.28]
- Install method: [helm, static manifests]
- Install flavor: [custom]

**Additional context**

Also I have fetched content of grastate.dat from one of the failing instances:
```bash
cat /var/lib/mysql/grastate.dat
# GALERA saved state
version: 2.1
uuid:    c045a915-378c-11ef-848a-ab7c4e33bf0b
seqno:   -1
safe_to_bootstrap: 0
```
According to Galera docs:
https://galeracluster.com/library/documentation/crash-recovery.html
it seems that it's a "All Nodes Go Down Without a Proper Shutdown Procedure" case and operator have some problems with stable recovery in this scenario.

[mariadb_galera_error.log](https://github.com/user-attachments/files/16054191/mariadb_galera_error.log)
[mariadb_galera_agent.log](https://github.com/user-attachments/files/16054212/mariadb_galera_agent.log)
[mariadb_galera_init.log](https://github.com/user-attachments/files/16054213/mariadb_galera_init.log)
[mariadb_galera_operator_part.log](https://github.com/user-attachments/files/16054245/mariadb_galera_operator_part.log)

**Comment by mmontes11** [comment]
Hey there @PeterStuck ! Thanks a lot for the detailed information.

I'm debugging what seems to be a similar issue here, please take a look: https://github.com/mariadb-operator/mariadb-operator/issues/672#issuecomment-2210499152

> changes on network

Please bear in mind that the operator needs to fetch the sequence number by making HTTP calls to the Galera agents. All the sequence numbers need to be obtained before a new cluster is bootstrapped, otherwise we might be causing data loss by bootstrapping a new cluster in a node that is not up to date.

**Comment by mmontes11** [comment]
We managed solve the problem in this related issue, we have described the steps to follow in-depth. Please let me know if this works for you: https://github.com/mariadb-operator/mariadb-operator/issues/669

TL;DR
- Set `podSecurityContext` in `MariaDB`:
```yaml
  podSecurityContext:
    runAsUser: 999
    runAsGroup: 999
    fsGroup: 999
```
- If the CRI/kubelet does not change the permissions, delete the PVC from the `Pods` that can't be recovered.

**Comment by PeterStuck** [comment]
Hi @mmontes11 thanks for taking our time on this case!

I changed security permissions like you proposed, extended podRecoveryTimeout to 5m, but it unfortunetely doesn't fix this problem.

After simple pod deletion:
```bash
kubectl delete pods -l app.kubernetes.io/instance=mariadb-galera
```

All my instances are stuck in same place in logs:
```bash
Defaulted container "mariadb" out of: mariadb, agent, init (init)
2024-07-08 09:43:28+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.11.7+maria~ubu2204 started.
2024-07-08 09:43:29+00:00 [Note] [Entrypoint]: MariaDB upgrade information missing, assuming required
2024-07-08 09:43:29+00:00 [Note] [Entrypoint]: MariaDB upgrade (mariadb-upgrade or creating healthcheck users) required, but skipped due to $MARIADB_AUTO_UPGRADE setting
mariadbd: Can't create file '/dev/stdout.err' (errno: 13 "Permission denied")
2024-07-08  9:43:29 0 [Note] Starting MariaDB 10.11.7-MariaDB-1:10.11.7+maria~ubu2204-log source revision 87e13722a95af5d9378d990caf48cb6874439347 as process 1
2024-07-08  9:43:29 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2024-07-08  9:43:29 0 [Note] InnoDB: Number of transaction pools: 1
2024-07-08  9:43:29 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2024-07-08  9:43:29 0 [Note] mariadbd: O_TMPFILE is not supported on /tmp (disabling future attempts)
2024-07-08  9:43:29 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
2024-07-08  9:43:29 0 [Note] InnoDB: Completed initialization of buffer pool
2024-07-08  9:43:29 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
2024-07-08  9:43:29 0 [Note] InnoDB: End of log at LSN=2353585
2024-07-08  9:43:29 0 [Note] InnoDB: 128 rollback segments are active.
2024-07-08  9:43:29 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2024-07-08  9:43:29 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2024-07-08  9:43:29 0 [Note] InnoDB: log sequence number 2353585; transaction id 5362
2024-07-08  9:43:29 0 [Warning] InnoDB: Skipping buffer pool dump/restore during wsrep recovery.
2024-07-08  9:43:29 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-07-08  9:43:29 0 [Note] Server socket created on IP: '0.0.0.0'.
2024-07-08  9:43:29 0 [Note] Server socket created on IP: '::'.
2024-07-08  9:43:29 0 [Note] WSREP: Recovered position: 7855ea03-3d0d-11ef-b11c-6b62aaac85b0:2608
```

On cluster restart case I'm getting slighty different log (recovered position part):
```bash
2024-07-08 11:16:50+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.11.7+maria~ubu2204 started.
2024-07-08 11:16:51+00:00 [Note] [Entrypoint]: MariaDB upgrade information missing, assuming required
2024-07-08 11:16:51+00:00 [Note] [Entrypoint]: MariaDB upgrade (mariadb-upgrade or creating healthcheck users) required, but skipped due to $MARIADB_AUTO_UPGRADE setting
mariadbd: Can't create file '/dev/stdout.err' (errno: 13 "Permission denied")
2024-07-08 11:16:51 0 [Note] Starting MariaDB 10.11.7-MariaDB-1:10.11.7+maria~ubu2204-log source revision 87e13722a95af5d9378d990caf48cb6874439347 as process 1
2024-07-08 11:16:51 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2024-07-08 11:16:51 0 [Note] InnoDB: Number of transaction pools: 1
2024-07-08 11:16:51 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2024-07-08 11:16:51 0 [Note] mariadbd: O_TMPFILE is not supported on /tmp (disabling future attempts)
2024-07-08 11:16:51 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
2024-07-08 11:16:51 0 [Note] InnoDB: Completed initialization of buffer pool
2024-07-08 11:16:51 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
2024-07-08 11:16:51 0 [Note] InnoDB: End of log at LSN=2483216
2024-07-08 11:16:51 0 [Note] InnoDB: 128 rollback segments are active.
2024-07-08 11:16:51 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2024-07-08 11:16:51 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2024-07-08 11:16:51 0 [Note] InnoDB: log sequence number 2483216; transaction id 5384
2024-07-08 11:16:51 0 [Warning] InnoDB: Skipping buffer pool dump/restore during wsrep recovery.
2024-07-08 11:16:51 0 [Note] Plugin 'FEEDBACK' is disabled.
2024-07-08 11:16:51 0 [Note] Server socket created on IP: '0.0.0.0'.
2024-07-08 11:16:51 0 [Note] Server socket created on IP: '::'.
2024-07-08 11:16:51 0 [Note] WSREP: Recovered position: 00000000-0000-0000-0000-000000000000:0
```

There are no error logs in agent and init containers. Seems that the recovery feature works but it's not stable based on my observation. To further analyze this let's sync mariadb config:
```bash
# https://github.com/mariadb-operator/mariadb-operator/blob/main/docs/API_REFERENCE.md
apiVersion: k8s.mariadb.com/v1alpha1
kind: MariaDB
metadata:
  name: mariadb-galera
  namespace: ovoo-rvms
spec:
  rootPasswordSecretKeyRef:
    name: mariadb
    key: root-password
    generate: true

  image: mariadb:10.11.7
  imagePullPolicy: IfNotPresent

  replicas: 3 # always an odd number of nodes

  galera:
    enabled: true
    primary:
      podIndex: 0
      automaticFailover: true
    replicaThreads: 1
    agent:
      image: ghcr.io/mariadb-operator/mariadb-operator:v0.0.28 # refer to appVersion in Chart.yaml for mariadb-operator
      port: 5555
      kubernetesAuth:
        enabled: true
      gracefulShutdownTimeout: 1s
    recovery:
      enabled: true
      minClusterSize: 50%
      clusterMonitorInterval: 10s
      clusterHealthyTimeout: 30s
      clusterBootstrapTimeout: 10m
      podRecoveryTimeout: 5m
      podSyncTimeout: 5m
    config:
      reuseStorageVolume: false
      volumeClaimTemplate:
        resources:
          requests:
            storage: 1Gi
        accessModes:
          - ReadWriteOnce

  port: 3306

  storage:
    size: 1Gi
    storageClassName: microk8s-hostpath
    resizeInUseVolumes: true
    waitForVolumeResize: true
    volumeClaimTemplate:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 1Gi
      storageClassName: microk8s-hostpath
      volumeMode: Filesystem

  myCnf: |
    [mariadb]
    bind-address=*
    default_storage_engine=InnoDB
    binlog_format=row
    innodb_autoinc_lock_mode=2
    max_allowed_packet=256M
    
    general_log = 1
    #general_log_file = /var/log/mysql/mysql.log
    general_log_file = /dev/stdout
    
    #log_error = /var/log/mysql/error.log
    log_error = /dev/stdout

  resources:
    requests:
      cpu: 64m
      memory: 128M
    limits:
      cpu: 256m
      memory: 1Gi

  env:
    - name: MARIADB_ROOT_PASSWORD
      valueFrom:
        secretKeyRef:
          name: mariadb
          key: root-password

  podSecurityContext:
#    runAsUser: 0
    runAsUser: 999
    runAsGroup: 999
    fsGroup: 999

#  securityContext:
#    allowPrivilegeEscalation: false

  livenessProbe:
    exec:
      command:
        - bash
        - -c
        - mysql -u root -p"${MARIADB_ROOT_PASSWORD}" -e "SELECT 1;"
    initialDelaySeconds: 20
    periodSeconds: 5
    timeoutSeconds: 5

  readinessProbe:
    exec:
      command:
        - bash
        - -c
        - mysql -u root -p"${MARIADB_ROOT_PASSWORD}" -e "SELECT 1;"
    initialDelaySeconds: 20
    periodSeconds: 5
    timeoutSeconds: 5

  affinity:
    antiAffinityEnabled: false
    # todo enable conditionally
#    podAntiAffinity:
#      requiredDuringSchedulingIgnoredDuringExecution:
#        - labelSelector:
#            matchLabels:
#              app.kubernetes.io/instance: mariadb-galera
#          topologyKey: kubernetes.io/hostname

  podDisruptionBudget:
    maxUnavailable: 50%

  updateStrategy:
    type: RollingUpdate

  service:
    type: LoadBalancer
    metadata:
      annotations:
        metallb.universe.tf/loadBalancerIPs: 172.18.0.20
    externalTrafficPolicy: Cluster
    sessionAffinity: None

  primaryService:
    type: LoadBalancer
    metadata:
      annotations:
        metallb.universe.tf/loadBalancerIPs: 172.18.0.160

  secondaryService:
    type: LoadBalancer
    metadata:
      annotations:
        metallb.universe.tf/loadBalancerIPs: 172.18.0.161

  metrics:
    enabled: true
    exporter:
      image: prom/mysqld-exporter:v0.15.1
      port: 9104
    serviceMonitor:
      jobLabel: system-monitoring-prometheus
      interval: 15s
      scrapeTimeout: 15s
```

Again thanks a lot for your input in this problem!

**Comment by mmontes11** [comment]
@PeterStuck Thanks again for the detailed response and the logs! 

> WSREP: Recovered position: 00000000-0000-0000-0000-000000000000:0

Judging by this, it seems like I've managed to reproduce a similar if not the same problem, see:
- https://github.com/mariadb-operator/mariadb-operator/issues/672#issuecomment-2236251146

**Comment by PeterStuck** [comment]
@mmontes11 I think if proposal in #672 will fix the issue we can close that one.

**Comment by mmontes11** [comment]
Thanks @PeterStuck ! Closing!
